pkg/executables/cmk.go

```diff
@@ -119,39 +139,136 @@ func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domain string,
 	return nil
 }
 
-func (c *Cmk) ValidateZonePresent(ctx context.Context, zone v1alpha1.CloudStackResourceRef) error {
+func (c *Cmk) ValidateZonesPresent(ctx context.Context, zones []v1alpha1.Zone) error {
 	var filterArg string
-	if zone.Type == v1alpha1.Id {
-		filterArg = fmt.Sprintf("id=\"%s\"", zone.Value)
+
+	for _, z := range zones {
+		zone := z.ZoneRef
+		if zone.Type == v1alpha1.Id {
+			filterArg = fmt.Sprintf("id=\"%s\"", zone.Value)
+		} else {
+			filterArg = fmt.Sprintf("name=\"%s\"", zone.Value)
+		}
+		result, err := c.exec(ctx, "list", "zones", filterArg)
+		if err != nil {
+			return fmt.Errorf("error getting zones info: %v", err)
+		}
+		if result.Len() == 0 {
+			return fmt.Errorf("zone %s not found", filterArg)
+		}
+
+		response := struct {
+			CmkZones []cmkZone `json:"zone"`
+		}{}
+		if err = json.Unmarshal(result.Bytes(), &response); err != nil {
+			return fmt.Errorf("failed to parse response into json: %v", err)
+		}
+		cmkZones := response.CmkZones
+		if len(cmkZones) > 1 {
+			return fmt.Errorf("duplicate zone %s found", cmkZones)
+		} else if len(zones) == 0 {
+			return fmt.Errorf("zone %s not found", filterArg)
+		} else {
+			c.zones = append(c.zones, cmkZones[0])
+		}
+	}
+
+	return nil
+}
+
+func (c *Cmk) ValidateNetworkPresent(ctx context.Context, domainId string, zone v1alpha1.Zone, account string, multipleZone bool) error {
```

Comment:
Is it possible to specify multiple zones with different networks?

Reply:
yes, each zone can have its network which can be different with network from other zones.